### PR TITLE
Fixed linux compilation issues

### PR DESCRIPTION
**Pull request by sthilaid**
Fixed the errors I encountered while building on linux.

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -348,3 +348,24 @@ MigrationBackup/
...
+HPL2/dependencies/
+Launcher.bin.x86_64
+Makefile
+cmake_install.cmake
+
+#ignore temp files
+
+
+
+
```
Comment (SuperSandro2000) [inline .gitignore:371]:
```suggestion

```

File: HPL2/core/include/BuildID_HPL2_0.h [hunk 2]
```diff
@@ -1,31 +1,12 @@
-/*
```
Comment (SuperSandro2000) [inline HPL2/core/include/BuildID_HPL2_0.h:1]: You shouldn't remove the copyright notice.

File: HPL2/core/sources/impl/PhysicsBodyNewton.cpp [hunk 3]
```diff
@@ -106,7 +106,9 @@ namespace hpl {
 		if(cPhysicsBodyNewton::mbUseCallback==false) return;
 
 		cPhysicsBodyNewton *pRigidBody = static_cast<cPhysicsBodyNewton*>(apEntity);
-		NewtonBodySetMatrix(pRigidBody->mpNewtonBody, &apEntity->GetLocalMatrix().GetTranspose().m[0][0]);
+
+        const cMatrixf matrix = apEntity->GetLocalMatrix().GetTranspose();
+        NewtonBodySetMatrix(pRigidBody->mpNewtonBody, &matrix.m[0][0]);
```
Comment (SuperSandro2000) [inline HPL2/core/sources/impl/PhysicsBodyNewton.cpp:111]:
```suggestion
		const cMatrixf matrix = apEntity->GetLocalMatrix().GetTranspose();
		NewtonBodySetMatrix(pRigidBody->mpNewtonBody, &matrix.m[0][0]);
```

Tabs

File: HPL2/core/sources/impl/PhysicsWorldNewton.cpp [hunk 4]
```diff
@@ -720,8 +720,9 @@ namespace hpl {
 		gDebugColor = aColor;
 
 		cCollideShapeNewton *pNewtonShape = static_cast<cCollideShapeNewton*>(apShape);
+        const cMatrixf transposeMatrix = a_mtxTransform.GetTranspose();
```
Comment (SuperSandro2000) [inline HPL2/core/sources/impl/PhysicsWorldNewton.cpp:723]:
```suggestion
		const cMatrixf transposeMatrix = a_mtxTransform.GetTranspose();
```

**Review by SuperSandro2000 — CHANGES_REQUESTED** [review]

File: HPL2/core/include/BuildID_HPL2_0.h (hunk 2 above)
Comment (sthilaid) [reply HPL2/core/include/BuildID_HPL2_0.h:1]: This file is auto generated... I'm not sure why the copyright notice went away :/

**Pull request closed without merging**